### ISSUE TITLE
Commands per second?

### ISSUE THREAD

**Issue by mapleleafs89** [mclarkk/lifxlan#127]
Not an issue, but before I go an purchase a LIFX lightstrip I am wanting to know how many commands can be issued to the lightstrip to control the various zones per second. I'm coming from hue which performs very well presumably due to the bridge. I'm hoping I could use this api to create some neat effects with the zones on the strip for when my team scores.

**Comment by hagemt** [comment]
From https://developers.meethue.com/support/:

> You can send commands to the lights too fast. If you stay roughly around 10 commands per second to the /lights resource as maximum you should be fine. For /groups commands you should keep to a maximum of 1 per second.

From https://lan.developer.lifx.com/#limits:

> Maximum recommended message transmit rate to a device: 20 per second

As of 2019, July 31st.

**Comment by DrTexx** [comment]
@mapleleafs89 @hagemt 
It may not be good for my bulbs, but I'm able to achieve ~400hz using rapid=True across all LIFX devices I have. Results are pretty consistant with my setup too.

Above ~500Hz most devices begin to really struggle and are visibly overwhelmed by the packets being sent to them. In my experience strips top out ~300Hz.

Multizone on strips however, is very very buggy if it's not run slow; in fact I'm not completely sure if rapid actually works at all. I've got the latest version of the LIFX Z (at time of posting) and I'm using lifxlan 1.2.5 (latest at the moment). I can barely achieve 20Hz when sending multizone packets to the strip.

Hope that can provide some (belated) insight :)

***TL;DR***
You can achieve ~300Hz on any LIFX device when setting overall color, however multizone packets top-out at ~20Hz for me and cause crashes a lot.

**Comment by mclarkk** [comment]
The `rapid=True` parameter just means lifxlan won't request (and wait for) an acknowledgement message from the bulb, which is what you want when writing rapid light show. There may be other factors limiting the multizone lights to the 20 packets/second rate, but the `rapid` flag is still working.

With the multizone lights, is it that lifxlan won't exceed 20 packets/second, or is it that the LIFX Z itself starts to struggle?

**Comment by DrTexx** [comment]
Oh sorry totally forgot to respond to this! :sweat_smile: 

It's that the multizone lights themselves begin to struggle, they get all blinky and laggy and clearly are struggling to keep up. I've experienced this with both the LIFX Tile and the LIFX Z Strip.
